### ISSUE TITLE
twisted.test.test_ssl.SpammyTLSTestCase exercises codepaths with warnings without suppressing them

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2055]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2055|
|Type|defect|
|Created|2006-09-02 20:23:00Z|

These tests should be fixed and un-todo'd, of course.  But they also shouldn't spam warnings until then.

<details><summary>Searchable metadata</summary>

```
trac-id__2055 2055
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1157228580000000 1157228580000000
changetime__1157565618000000 1157565618000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

```
Index: twisted/test/test_ssl.py
===================================================================
--- twisted/test/test_ssl.py    (revision 18030)
+++ twisted/test/test_ssl.py    (working copy)
@@ -3,7 +3,7 @@
 
 
 from __future__ import nested_scopes
-from twisted.trial import unittest
+from twisted.trial import unittest, util as trial_util
 from twisted.internet import protocol, reactor, interfaces, defer
 from twisted.protocols import basic
 from twisted.python import util, log, runtime
@@ -301,14 +301,28 @@
         )
 
 
+
+_bufferedSuppression = trial_util.suppress(
+    message="startTLS with unwritten buffered data currently doesn't work "
+            "right. See issue #686. Closing connection.",
+    category=RuntimeWarning)
+
+
 class SpammyTLSTestCase(TLSTestCase):
+    """
+    Test TLS features with bytes sitting in the out buffer.
+    """
     fillBuffer = 1
+
     def testTLS(self):
-        TLSTestCase.testTLS(self)
+        return TLSTestCase.testTLS(self)
+    testTLS.suppress = [_bufferedSuppression]
+    testTLS.todo = "startTLS doesn't empty buffer before starting TLS. :("
+
+
     def testBackwardsTLS(self):
-        TLSTestCase.testBackwardsTLS(self)
-
-    testTLS.todo = "startTLS doesn't empty buffer before starting TLS. :("
+        return TLSTestCase.testBackwardsTLS(self)
+    testBackwardsTLS.suppress = [_bufferedSuppression]
     testBackwardsTLS.todo = "startTLS doesn't empty buffer before starting TLS. :("
 

```

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_2"></a>|@jerub set owner to @exarkun|
|-|-|

Yuck. At least it cleans up the build. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [18062]) Apply patch from [[#2055](https://github.com/twisted/twisted/issues/2055)](https://github.com/twisted/twisted/issues/2055) - suppress runtime warnings in test suite

Author: exarkun
Reviewer: jerub
Fixes #2055

There are two tests marked TODO for starting a TLS handshake while there are
still bytes in Twisted's output buffer.  These fail, but also emit a RuntimeWarning.
This adds suppressions for these warnings to the test methods in question.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
